src/PowerShell/Microsoft.WinGet.Client/PSObjects/CatalogPackage.cs

```diff
@@ -0,0 +1,129 @@
+﻿// -----------------------------------------------------------------------------
+// <copyright file="CatalogPackage.cs" company="Microsoft Corporation">
+//     Copyright (c) Microsoft Corporation. Licensed under the MIT License.
+// </copyright>
+// -----------------------------------------------------------------------------
+
+namespace Microsoft.WinGet.Client.PSObjects
+{
+    using System.Linq;
+    using Microsoft.Management.Deployment;
+    using Microsoft.WinGet.Client.Exceptions;
+
+    /// <summary>
+    /// CatalogPackage wrapper object for displaying to PowerShell.
+    /// </summary>
+    public class CatalogPackage
+    {
+        private Management.Deployment.CatalogPackage catalogPackage;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CatalogPackage"/> class.
+        /// </summary>
+        /// <param name="catalogPackage">CatalogPackage COM object.</param>
+        public CatalogPackage(Management.Deployment.CatalogPackage catalogPackage)
+        {
+            this.catalogPackage = catalogPackage;
+        }
+
+        /// <summary>
+        /// Gets the name of the catalog package.
+        /// </summary>
+        public string Name
+        {
+            get
+            {
+                return this.catalogPackage.Name;
+            }
+        }
+
+        /// <summary>
+        /// Gets the id of the catalog package.
+        /// </summary>
+        public string Id
+        {
+            get
+            {
+                return this.catalogPackage.Id;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the version of the catalog package.
+        /// </summary>
+        public string Version { get; protected set; }

```

Comment:
Where is Version set?

Reply:
spoke offline, made CatalogPackage and Version abstract class and method. 

Also changed catalogPackage to CatalogPackageCOM and made it a protected readonly property.